consensus/misc/eip1559.go

```diff
@@ -107,8 +107,71 @@ func CalcBaseFee(config *params.ChainConfig, parent *types.Header, parentL1BaseF
 		return big.NewInt(10000000) // 0.01 Gwei
 	}
 
-	scalar, overhead := ReadL2BaseFeeCoefficients()
-	return calcBaseFee(scalar, overhead, parentL1BaseFee)
+	if parent == nil || parent.Number == nil || !config.IsFeynman(parent.Time) {
+		scalar, overhead := ReadL2BaseFeeCoefficients()
```

Comment:
Can we somehow consolidate the contracts? I'm a bit confused what is stored in which contract currently and what it is used for.